src/controllers/helpers.js

```diff
@@ -0,0 +1,104 @@
+const {
+  Author, Book, Reader, Genre,
+} = require('../models');
+
+const get404Error = (model) => ({ error: `The ${model} could not be found.` });
+
+const getModel = (model) => {
+  const models = {
+    book: Book,
+    reader: Reader,
+    genre: Genre,
+    author: Author,
+  };
+  return models[model];
+};
+
+const removePassword = (sentData) => {
+  delete sentData.password;
+
+  return sentData;
+};
+
+const chooseModelToInclude = (model) => {
+  if (model === 'book') return { include: [{ model: Genre }, { model: Author }] };
+  if (model === 'genre') return { include: Book };
+  if (model === 'author') return { include: Book };
+
+  return {};
+};
+
+const getAllItems = (res, model) => {
+  const Model = getModel(model);
+
+  Model.findAll({ ...chooseModelToInclude(model) }).then((items) => {
+    const itemsWithNoPassword = items.map((item) => removePassword(item.dataValues));
+    res.status(200).json(itemsWithNoPassword);
+  });
+};
+
+const createItem = (res, model, item) => {
+  const Model = getModel(model);
+
+  Model.create(item)
+    .then((newItemCreated) => {
+      const itemPopPassword = newItemCreated;
```

Comment:
to make sure there is no connection use the spread operator like const itemPopPassword = {...newItemCreated}